### PR TITLE
Adjusted checkout to avoid OpenSSF false positive

### PR DESCRIPTION
**Pull request by eddie-knight**
Part of https://github.com/cilium/cilium/issues/21760

Signed-off-by: Eddie Knight <knight@linux.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #issue-number

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

**Comment by eddie-knight** [comment]
Validated that the failing OpenSSF Scorecard check is caused by a search for `ref` in the checkouts action.
 https://github.com/ossf/scorecard/blob/78c7e83b543164a30541ba45b9901b76eaf5039a/checks/raw/dangerous_workflow.go#L186-L189

File: .github/workflows/build-images-base.yaml
Comment (aanm) [inline .github/workflows/build-images-base.yaml:57]: @eddie-knight Thanks for the PR. Can you explain what is the issue of using `ref: ${{ github.event.pull_request.head.sha }}`? This workflow is behind an environment, see line 23, which is only executed after being approved by a set of committers.

**Review by aanm — CHANGES_REQUESTED** [review]

File: .github/workflows/build-images-base.yaml
Comment (eddie-knight) [reply .github/workflows/build-images-base.yaml:57]: Honestly I'm not able to explain much from the OpenSSF scorecard aside from what the [docs](https://github.com/ossf/scorecard/blob/7cd6406aef0b80a819402e631919293d5eb6adcf/docs/checks.md#dangerous-workflow) say 😅

We could raise an issue on the repo if it's a cocnern... but IMHO it seems like changing to just use the top commit on actions/checkout accomplishes the same thing as was here before, and we don't have to fuss with reforming the scorecard.

Comment (aanm) [reply .github/workflows/build-images-base.yaml:57]: @eddie-knight not specifying the right `ref`, in this case `${{ github.event.pull_request.head.sha }}`, we will default to the ref of the base branch which will not contain the changes submitted by a PR that will trigger this workflow.

Comment (eddie-knight) [reply .github/workflows/build-images-base.yaml:57]: @aanm Since you're using `actions/checkout` on a `pull_request` workflow, the default ref is the "Last merge commit on the GITHUB_REF branch"

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Comment (nbusseneau) [reply .github/workflows/build-images-base.yaml:57]: @eddie-knight This workflow uses `pull_request_target`, so `GITHUB_REF` points to the PR target branch, not the PR branch, hence why we need this ;)

Comment (eddie-knight) [reply .github/workflows/build-images-base.yaml:57]: Ah, I see my oversight. In that case I'll modify this PR to create a CLOMonitor exception instead.

Comment (eddie-knight) [reply .github/workflows/build-images-base.yaml:57]: Ah, I did overlook that detail. So our options from here are...

1. Propose a change in the Scorecard logic
2. Create a CLOMonitor exception
3. Reconsider the use of `pull_request_target`, per the [docs recommendation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) "Avoid using this event if you need to build or run code from the pull request."

Comment (aanm) [reply .github/workflows/build-images-base.yaml:57]: For 3. it will mean that external contributors aren't able to use this GH workflow, which is not intended. In this case it is safe to run `pull_request_target` since this workflow is only permitted to be executed after an explicit approval of a subset of committers.

Comment (eddie-knight) [reply .github/workflows/build-images-base.yaml:57]: Brilliant. I'll close out this PR and recommend you do #1 or #2 above... either one will require a bit of language you'll have to craft from your perspective.

Godspeed!

Comment (xmulligan) [reply .github/workflows/build-images-base.yaml:57]: This is fixed in #22371

**Pull request closed without merging**
